terraform/shared/ci-mirrors-access.tf

```diff
@@ -0,0 +1,35 @@
+resource "aws_iam_role" "foo" {
+  name = "ci--rust-lang--ci-mirrors"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = "sts:AssumeRoleWithWebIdentity"
+        Principal = {
+          Federated = aws_iam_openid_connect_provider.github_actions.arn
+        }
+        Condition = {
+          StringLike = {
+            "token.actions.githubusercontent.com:sub" = "repo:rust-lang/ci-mirrors:ref:refs/heads/gh-readonly-queue/main/*"
+          }
+        }
+      }
+    ]
+  })
+
+  inline_policy {
+    name = "permissions"
+    policy = jsonencode({
+      Version = "2012-10-17"
+      Statement = [
+        {
+          Effect   = "Allow"
+          Action   = ["s3:GetObject", "s3:PutObject"]
```

Comment:
Should we enforce never overwriting an object?

https://docs.aws.amazon.com/AmazonS3/latest/userguide/conditional-writes-enforce.html#conditional-writes-enforce-ex1

(This is pretty recently launched but seems like a good fit for our purposes here)

Reply:
That's amazing! Yes, we should definitely do that, I'll update the PR probably tomorrow.

Reply:
Added a policy to never overwrite the file.